app/models/concerns/source_identifier_filterable.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+# SourceIdentifierFilterable
+#
+# This module provides functionality for filtering records based on source identifiers.
+# A source identifier can be a plate barcode, tube barcode, or a combination of plate barcode
+# and well position.
+# Models that include this module will have methods to filter records using these identifiers.
+#
+# ## Methods:
+#
+# * apply_source_identifier_filter(records, value, plate_join: :plate,
+#                                  tube_join: :tube, well_join: :well)
+#   - Filters the given records based on the provided source identifiers.
+#   - Parameters:
+#     - records: The ActiveRecord relation to filter.
+#     - value: An array of source identifiers to filter by.
+#     - plate_join: The association name for joining with the plate table (default: :plate).
+#     - tube_join: The association name for joining with the tube table (default: :tube).
+#     - well_join: The association name for joining with the well table (default: :well).
+#
+# ## Example Usage:
+#
+#   class MyModel < ApplicationRecord
+#     include SourceIdentifierFilterable
+#   end
+#
+#   records = MyModel.all
+#   source_identifiers = ['PLATE123', 'TUBE456', 'PLATE789:A1']
+#   filtered_records = MyModel.apply_source_identifier_filter(records, source_identifiers)
+#
+#   # Custom join associations
+#   filtered_records = MyModel.apply_source_identifier_filter(records, source_identifiers,
+#                                                             plate_join: :source_plate,
+#                                                             tube_join: :source_tube)
+#
+module SourceIdentifierFilterable
+  extend ActiveSupport::Concern
+
+  class_methods do # rubocop:disable Metrics/BlockLength
+    def apply_source_identifier_filter(records, value, # rubocop:disable Metrics/MethodLength,Metrics/PerceivedComplexity
+                                       plate_join: :plate, tube_join: :tube, well_join: :well)
+      rec_ids = []
```

Comment:
personally I would use full names as there is no discernable difference from a syntactic view e.g record_ids, filtered_records. It removes ambiguity